packages/bitcore-wallet-client/lib/api.js

```diff
@@ -2401,6 +2159,50 @@ API.prototype.getWalletIdsFromOldCopay = function(username, password, blob) {
   return _.uniq(ids);
 };
 
-API.PayPro = PayPro;
+API.fromOld = function(x) {
+  $.shouldBeObject(x);
+
+  // TODO RO clients
+  if (!_.isUndefined(x.version) || (!x.xPrivKey && !x.xPrivKeyEncrypted)) {
+    throw 'Could not recognize old version';
+  }
+
+  var k = new Key();
+  _.each(Key.FIELDS, (i) => {
+    if (!_.isUndefined(x[i])) {
+      k[i] = x[i];
+    }
+  });
+
+  // ALL old credentials have multi wallets in 44'.
+  k.use48forMultisig = false;
+
+  let obsolteFields = {
+    version: true,
+    xPrivKey: true,
+    xPrivKeyEncrypted: true,
+    hwInfo: true,
+    entropySourcePath: true,
+    mnemonic: true,
+    mnemonicEncrypted: true,
+  };
 
+  var c = new Credentials();
+  _.each(Credentials.FIELDS, function(i) {
+    if (!obsolteFields[i]) {
+      c[i] = x[i];
+    }
+  });
+  if (c.externalSource) {
+    throw new Error('External Wallets are no longer supported');
+  }
+  c.coin = c.coin || 'btc';
+  c.addressType = c.addressType || Constants.SCRIPT_TYPES.P2SH;
+  c.account = c.account || 0;
+  c.rootPath = c.getRootPath();
```

Comment:
Here is missing this: `c.keyId = k.id`

Reply:
Should it also be uploaded to version 2? (`c.version = 2`) ?

Reply:
`new Credentials` returns `.version=2`  should it should work.
adding keyId.

Reply:
done.